lib/compress.js

```diff
@@ -1780,6 +1780,35 @@ merge(Compressor.prototype, {
         node.DEFMETHOD("has_side_effects", func);
     });
 
+    // determine if expression is constant
+    (function(def){
+        function all(list) {
+            for (var i = list.length; --i >= 0;)
+                if (!list[i].is_constant_expression())
+                    return false;
+            return true;
+        }
+        def(AST_Node, return_false);
+        def(AST_Constant, return_true);
+        def(AST_Unary, function(){
+            return this.expression.is_constant_expression();
+        });
+        def(AST_Binary, function(){
+            return this.left.is_constant_expression() && this.right.is_constant_expression();
+        });
+        def(AST_Array, function(){
+            return all(this.elements);
+        });
+        def(AST_Object, function(){
+            return all(this.properties);
+        });
+        def(AST_ObjectProperty, function(){
+            return this.value.is_constant_expression();
```

Comment:
For harmony:
```js
return !(this.key instanceof AST_Node) && this.value.is_constant_expression();
```
It should also work on master, although it would be slower.